flang/lib/Optimizer/Builder/Runtime/Command.cpp

```diff
@@ -19,3 +20,46 @@ mlir::Value fir::runtime::genCommandArgumentCount(fir::FirOpBuilder &builder,
       fir::runtime::getRuntimeFunc<mkRTKey(ArgumentCount)>(loc, builder);
   return builder.create<fir::CallOp>(loc, argumentCountFunc).getResult(0);
 }
+
+void fir::runtime::genGetCommandArgument(fir::FirOpBuilder &builder,
+                                         mlir::Location loc, mlir::Value number,
+                                         mlir::Value value, mlir::Value length,
+                                         mlir::Value status,
+                                         mlir::Value errmsg) {
+  auto argumentValueFunc =
+      fir::runtime::getRuntimeFunc<mkRTKey(ArgumentValue)>(loc, builder);
+  auto argumentLengthFunc =
+      fir::runtime::getRuntimeFunc<mkRTKey(ArgumentLength)>(loc, builder);
+
+  auto isPresent = [&](mlir::Value val) -> bool {
+    auto definingOp = val.getDefiningOp();
+    if (auto cst = mlir::dyn_cast<fir::AbsentOp>(definingOp))
+      return false;
+    return true;
+  };
```

Comment:
I'm not sure this is the correct way. In practice there is no issue. I tried using the "fir::IsPresentOp" but could not seem to get it working. Any idea if this is fine?

Reply:
Done.